pulp_container/app/registry_api.py

```diff
@@ -761,20 +737,8 @@ def put(self, request, path, pk=None):
                     ca.artifact = artifact
                     ca.save(update_fields=["artifact"])
         upload.delete()
-
-        dispatched_task = dispatch(
-            add_and_remove,
-            shared_resources=[f"upload:{pk}"],
-            exclusive_resources=[repository],
-            kwargs={
-                "repository_pk": str(repository.pk),
-                "add_content_units": [str(blob.pk)],
-                "remove_content_units": [],
-            },
-        )
-
-        if has_task_completed(dispatched_task):
-            return BlobResponse(blob, path, 201, request)
+        repository.pending_blobs.add(blob)
```

Comment:
since you should removed the task dispatch you should remove line 677 and instead of those check repo.pending.blobs containing the blob digest in question